src/styles/auto-prefix.js

```diff
@@ -3,61 +3,97 @@ import warning from 'warning';
 
 const prefixers = {};
 
+let hasWarnedAboutNavigator = false;
+
+function getPrefixer(userAgent) {
+  if (typeof navigator !== 'undefined') {
+    userAgent = navigator.userAgent;
+  }
+
+  if (userAgent === null && !hasWarnedAboutNavigator) {
+    warning(false, `Material-UI: userAgent should be supplied in the muiTheme context
+      for server-side rendering.`);
+
+    hasWarnedAboutNavigator = true;
+
+    return null;
+  }
+
+  // Get prefixing instance for this user agent
+  let prefixer = prefixers[userAgent];
```

Comment:
> I do think that it's a good idea to cache the prefixer.

Yes, I think I wasn't clear. I mean if we treat prefixer as a transformer builder. like:

```js
gettransformer(theme){
// build prefixer

  return (style) => {
    // use prefixer
  }
}
```

We'll create a prefixer only once per request! or once per theme creation to be more accurate.

Because creating a prefixer for every user-agent seems redundant. But we should definitely cache one prefixer per theme creating. that's likely to be once per request (or page load for client side rendering)